From 5c6373dc418179085cf116902a8601e58c295ea6 Mon Sep 17 00:00:00 2001
From: Sai Surya Teja <sssteja@habana.ai>
Date: Thu, 28 Oct 2021 08:37:27 +0300
Subject: [PATCH] [SW-57475] gaudi2: hard reset en if not skip fw

Enabling hard reset in preboot if the skip firmware config is not
set on reading from boot strap bit 11.

Change-Id: Id47e1f22438b47561c769fc88901f9fe2354d393
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index bcdadb5..4b4bceb 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1181,6 +1181,12 @@
 	return ((u32)base - SDRAM_BASE);
 }
 
+void lkd_dev_specific_setup(void)
+{
+	if (!hl_skip_fw_cfg())
+		cpu_boot_dev_sts_set(CPU_BOOT_DEV_STS_FW_HARD_RST_EN);
+}
+
 u16 hl_get_boot_info_ver(void)
 {
 	return HL_BOOT_INFO_VERSION;
